### PR TITLE
Added example for stream mode

### PR DESCRIPTION
**Pull request by fanantoxa**
It was a bit hard to understand how to work with stream mode so I've added example for stream mode.

### CODE REVIEW

**Comment by fanantoxa** [comment]
@jnunemaker, all tests except on ruby 1.9.3 pass.
Cloud you please merge example to master?
Also added issue here: https://github.com/jnunemaker/httparty/issues/486

File: examples/stream_download.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+dir = File.expand_path(File.join(File.dirname(__FILE__), '..', 'lib'))
+require File.join(dir, 'httparty')
+require 'pp'
+
+class StreamDownload
+  include HTTParty
+  stream_body = true
```
Comment (jnunemaker) [inline examples/stream_download.rb]: I don't believe this is correct. [stream_body is an option for requests](https://github.com/jnunemaker/httparty/blob/0d808c7f3a68b8d091506e22003fc00dd84bf6a6/spec/httparty_spec.rb#L777-L788), not a class level method. Also, class level methods don't use ='s as that would mean assignment, not a method call.

**Comment by fanantoxa** [comment]
@jnunemaker I've updated code with your additions

**Comment by fanantoxa** [comment]
@jnunemaker please take a look when you'll find time please )

**Pull request merged**